benches/sync_mpsc_oneshot.rs

```diff
@@ -1,27 +1,28 @@
-use bencher::{benchmark_group, benchmark_main, Bencher};
 use tokio::{
     runtime::Runtime,
     sync::{mpsc, oneshot},
 };
 
-fn request_reply_current_thread(b: &mut Bencher) {
+use criterion::{criterion_group, criterion_main, Criterion};
+
+fn request_reply_current_thread(c: &mut Criterion) {
     let rt = tokio::runtime::Builder::new_current_thread()
         .build()
         .unwrap();
 
-    request_reply(b, rt);
+    request_reply(c, rt);
 }
 
-fn request_reply_multi_threaded(b: &mut Bencher) {
+fn request_reply_multi_threaded(c: &mut Criterion) {
     let rt = tokio::runtime::Builder::new_multi_thread()
         .worker_threads(1)
         .build()
         .unwrap();
 
-    request_reply(b, rt);
+    request_reply(c, rt);
 }
 
-fn request_reply(b: &mut Bencher, rt: Runtime) {
+fn request_reply(b: &mut Criterion, rt: Runtime) {
```

Comment:
Would it be better to make the `request_reply` has a name of runtime, such as `muilti_thread` and `current_thread`? Or we got `request_reply` and `request_reply #2`, they are not very clear name of benchmark test.

Reply:
I focused primarily on replacing `bencher` with `criterion` in this PR. I've attempted to incorporate some of the clean-up suggestions you provided, but I'd like to concentrate on other issues at the moment. If there are additional clean-ups you'd like to see, it might be beneficial to address them in a subsequent PR.

Reply:
I am ok with it. This PR looks good to me. Thanks.
@Darksonn could you review this?